spacelift/resource_stack.go

```diff
@@ -914,13 +847,22 @@ func ignoreOnceCreated(_, _, _ string, d *schema.ResourceData) bool {
 }
 
 func resourceStackImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-	if d.Id() == "" {
+	stackID := d.Id()
+	if stackID == "" {
 		return nil, errors.New("stack ID is required to import a stack")
 	}
 
-	diag := resourceStackRead(ctx, d, meta)
-	if diag.HasError() {
-		return nil, fmt.Errorf("could not import stack: %v", diag)
+	stack, err := getStackByID(ctx, meta.(*internal.Client), stackID)
+	if err != nil {
+		return nil, fmt.Errorf("Could not query for stack with ID %q: %v", stackID, err)
+	}
+
+	if stack == nil {
```

Comment:
Would it make sense to also suggest that the stack may exist but you have no access to it?

Reply:
Sure - I'll update the message.

Reply:
Ok - that's the message updated.